### PR TITLE
[#93455] Ensure auto-logout instruments don't complete unpurchased orders

### PR DESCRIPTION
**Pull request by jhanggi**

### CODE REVIEW

File: app/models/order_detail.rb [hunk 1]
```diff
@@ -129,6 +129,13 @@ def self.for_facility_url(facility_url)
     details
   end
 
+  def self.purchased_active_reservations
+    where(state: ['new', 'inprocess'])
+      .where('order_status_id IS NOT NULL')
+      .where(reservations: { canceled_at: nil})
```
Comment (drench) [inline app/models/order_detail.rb]: space after `nil`?

File: app/models/order_detail.rb [hunk 2]
```diff
@@ -129,6 +129,13 @@ def self.for_facility_url(facility_url)
     details
   end
 
+  def self.purchased_active_reservations
+    where(state: ['new', 'inprocess'])
```
Comment (drench) [inline app/models/order_detail.rb]: The combo of `new` and `inprocess` is a recurring theme in this model. I don't know if you want to deal with DRYing them all up now but I want to at least keep it in mind as a TODO.

File: app/support/auto_logout.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+class AutoLogout
```
Comment (drench) [inline app/support/auto_logout.rb]: I realize this class isn't "yours" but there are parts of this extraction that could use cleanup (despite Code Climate giving it an "A").

File: spec/factories/reservations.rb [hunk 4]
```diff
@@ -2,6 +2,10 @@
   factory :reservation do
     reserve_start_at { Time.zone.parse("#{Date.today.to_s} 10:00:00") + 1.day }
     reserve_end_at { reserve_start_at + 1.hour }
+
+    trait :yesterday do
+      reserve_start_at { Time.zone.parse("#{Date.today.to_s} 10:00:00") - 1.day }
```
Comment (drench) [inline spec/factories/reservations.rb]: While this is consistent with the rest of the file, is the `#to_s` necessary?

**Comment by drench** [comment]
Just had a few questions and notes.

File: app/models/order_detail.rb (hunk 2 above)
Comment (jhanggi) [inline app/models/order_detail.rb]: Yeah, I agree. That was something I didn't really want to take on with this fix right now, especially as some of them would involve full refactoring of the old `scope`/`condition:` syntax.

File: app/support/auto_logout.rb (hunk 3 above)
Comment (jhanggi) [inline app/support/auto_logout.rb]: I agree. There's also some shared code with the auto cancel job. That was more than I wanted to take on with this ticket. I think a good time to clean this class up further will be when we address why it's not actually toggling the instrument off and the other tickets Aaron mentioned in the ticket.

**Comment by jhanggi** [comment]
I made a couple of the minor tweaks. The other refactorings I didn't necessarily want to take on right now, since there weren't any tests around this feature to provide a comfortable safety harness. I had already spent a good amount of time trying to track down this issue, and I'm still not 100% positive that this is the root cause. I tried to make things a little better and set up some of the future refactorings.

**Comment by drench** [comment]
:+1:

**Pull request merged**